### PR TITLE
Add request tests for tags and topics

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/wgkfOJsE/1283-improve-consistency-in-content-publisher-tests

This follows on the test strategy introduced in #1590. Towards this I've done a bunch of writing of request tests and deleting of feature tests. Rather than do one large PR I thought it was best to separate into distinct PR's that are easier to review and thought I'd start with one that covers a few.

This PR adds request tests for tags and topic endpoints and removes the dead-end feature tests that are now covered by the request tests.

### CODE REVIEW

File: spec/requests/tags_spec.rb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      stub_publishing_api_has_linkables(
+        [{ "content_id" => current_user.organisation_content_id,
+           "internal_name" => "Organisation" }],
+        document_type: tag_field.document_type,
+      )
+
+      get tags_path(edition.document)
+      expect(response).to have_http_status(:ok)
+      expect(response.body)
+        .to have_content(I18n.t!("tags.edit.organisation_warning"))
```
Comment (benthorner) [inline spec/requests/tags_spec.rb]: Thinking back to our discussion on the different types of test: would this be better suited (and perhaps more easily implemented) in a view test? It seems to be more focussed on testing a property of the view, than the nature of the request/response.

File: spec/requests/tags_spec.rb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    end
+  end
+
+  describe "PATCH /documents/:document/tags" do
+    before { stub_any_publishing_api_put_content }
+
+    it "redirects to document path on successful request" do
+      tag_field = build(:tag_field, type: "multi_tag", id: "field")
```
Comment (benthorner) [inline spec/requests/tags_spec.rb]: Minor: do we need any tag fields for this to work?

File: spec/requests/tags_spec.rb (hunk 1 above)
Comment (kevindew) [reply spec/requests/tags_spec.rb]: Yes, you're right - I've moved this into a view test.

File: spec/requests/tags_spec.rb (hunk 2 above)
Comment (kevindew) [reply spec/requests/tags_spec.rb]: No, it doesn't I've removed this. Thanks

**Pull request merged**